spec/models/submission_file_spec.rb

```diff
@@ -0,0 +1,314 @@
+require 'spec_helper'
+
+describe SubmissionFile do
+  # Ensure that the the following relationship exists
+  context 'An existing SubmissionFile' do
+    it { is_expected.to belong_to(:submission) }
+    it { is_expected.to have_many(:annotations) }
+    it { is_expected.to validate_presence_of :submission }
+    it { is_expected.to validate_presence_of :filename }
+    it { is_expected.to validate_presence_of :path }
+  end
+
+  context 'A SubmissionFile' do
+
+    context 'A SubmissionFile without parameter and without id' do
+      before(:each) do
+        @submissionfile = SubmissionFile.create()
+      end
+      it 'be invalid and should not be saved' do
+        expect(@submissionfile).not_to be_valid
+        expect(@submissionfile.save).to be false
+      end
+    end
+
+    context 'A SubmissionFile without parameter and with id' do
+      before(:each) do
+        @submissionfile = SubmissionFile.create()
+        @submissionfile.submission_id = 1
+      end
+      it 'be invalid and should not be saved' do
+        expect(@submissionfile).not_to be_valid
+        expect(@submissionfile.save).to be false
+      end
+    end
+
+    context 'A SubmissionFile without filename' do
+      before(:each) do
+        @submissionfile = SubmissionFile.create(filename: '',
+                                                path:     'path')
+        @submissionfile.submission_id = 1
+      end
+
+      it 'be invalid and should not be saved' do
+        expect(@submissionfile).not_to be_valid
+        expect(@submissionfile.save).to be false
+      end
+    end
+
+    context 'A SubmissionFile without path' do
+      before(:each) do
+        @submissionfile = SubmissionFile.create(filename: 'filename',
+                                                path:     '')
+        @submissionfile.submission_id = 1
+      end
+
+      it 'be invalid and should not be saved' do
+        expect(@submissionfile).not_to be_valid
+        expect(@submissionfile.save).to be false
+      end
+    end
+
+    context 'A SubmissionFile with filename and path, but without id' do
+      before(:each) do
+        @submissionfile = SubmissionFile.create(filename: 'filename',
+                                                path:     'path')
+      end
+
+      it 'be invalid and should not be saved' do
+        expect(@submissionfile).not_to be_valid
+        expect(@submissionfile.save).to be false
+      end
+    end
+
+    context 'A .java Submission file' do
+      before(:each) do
+        @submissionfile = SubmissionFile.create(filename: 'filename.java',
+                                                path:     'path')
+        @submissionfile.submission_id = 1
```

Comment:
Please add this as a parameter to `create` (rather than a separate line of code).